l10n_es_aeat_silicie/models/stock_move.py

```diff
@@ -0,0 +1,683 @@
+                    if not move.silicie_loss_id:
+                        move.fields_check = False
+                    elif abs(a15_qty + move.absolute_alcohol - a14_qty) > 0.005:
+                        move.fields_check = False
+                        if a14_move:
+                            a14_move.write({"fields_check": False})
+                        if a15_move:
+                            a15_move.fields_check = False
+            else:
+                move.fields_check = False
+
+    @api.model
+    def _get_wsdl_params(self):
+        config_object = self.env["ir.config_parameter"].sudo()
+        result = {}
+        for item in SILICIE_WDSL_MAPPING.keys():
+            result[item] = config_object.get_param(
+                SILICIE_WDSL_MAPPING[item], False)
+        return result
+
+    @api.model
+    def _get_silicie_header(self):
+        cae = self.env.user.company_id.cae
+        if self.env.user.company_id.silicie_test:
+            cae = self.env.user.company_id.cae_for_tests
+        return {
+            "DatosEstablecimiento":
+                {
+                    "NIFEs": self.env.user.company_id.vat.replace("ES", ""),
+                    "CAEEs": cae,
+                }
+        }
+
+    @api.multi
+    def _send_aeat(self):
+        soap_obj = self.env["l10n.es.aeat.soap"]
+        params = self._get_wsdl_params()
+        header = self._get_silicie_header()
+        message_id = self.env["ir.sequence"].next_by_code(
+            "webservice.send.id")
+        header.update({"IdentificadorMensaje": message_id})
+        asientos = []
+        for move in self:
+            if move.send_silicie or move.not_declare:
+                continue
+            move.silicie_product_type = \
+                move.product_id.product_tmpl_id.silicie_product_type
+            if move.silicie_product_type:
+                datos_envase = ""
+                if move.silicie_product_type == "alcohol":
+                    tipo_envase = move.container_type_silicie_id.code or ""
+                    capacidad_envase = move.factor_conversion_silicie
+                    numero_envases = round(move.quantity_done)
+                    if not tipo_envase:
+                        capacidad_envase = ""
+                        numero_envases = ""
+                    datos_envase = {
+                        "TipoEnvase": tipo_envase,
+                        "CapacidadEnvase": capacidad_envase,
+                        "NumeroEnvases": numero_envases,
+                    }
+                datos_alcohol = {
+                    "GradoAlcoholico": move.alcoholic_grade,
+                    "CantidadAbsoluta": move.absolute_alcohol,
+                    "DatosEnvase": datos_envase,
+                }
+                datos_producto = {
+                    "CodigoEpigrafe": move.epigraph_silicie_id.code or "",
+                    "Clave": move.product_key_silicie_id.code or "",
+                    "CodigoNC": move.nc_code,
```

Comment:
La longitud tiene que ser <=8:
`"CodigoNC": move.nc_code and move.nc_code[:8],`